openqaoa/optimizers/result.py

```diff
@@ -301,3 +303,46 @@ def lowest_cost_bitstrings(self, n_bitstrings: int = 1) -> dict:
             ]
         }
         return best_results
+
+
+    def dumps(self, string=False, indent=4):
+        """
+        Returns the result serialized 
+
+        Parameters
+        ----------
+        string : bool
+            If True, the result is returned as a string. Otherwise, it is returned as a dictionary.
+
+        Returns
+        -------
+        str or dict
+        """
+
+        if string:
+            return json.dumps(convert2serialize_complex(self), indent=indent)
+        else:
+            return convert2serialize(self)
+
+    def dump(self, filename=None, indent=4):
+        """
+        Saves the result as json file.
+
+        Parameters
+        ----------
+        filename : str
+            The name of the file to save the result. If None, the result is saved as 'result.json'.
+        """
+
+        # default filename 
+        if filename is None:
```

Comment:
Maybe we should just request to add the file name: otherwise you may end up running an experiment and overwriting all your results!